### PR TITLE
120 write vitals tests

### LINKED ISSUE #120: Unit tests for vitals
## Description

Currently there are no tests for vitals. We intent to add `VitalsForm` unit tests and to edit to use the neutral information toast notification instead of the error toast notification when no form fields are changed in the `VitalsForm`

## Acceptance Criterias
- [ ] Unit tests are added
- [ ] Submitting the form without changing any field shows a neutral (info) toast notification with the message "No form field changed!" instead of error toast
- [ ] Submitting the form after changing at least one field triggers the save mutation and shows a "Vitals saved successfully!" toast

### PR DESCRIPTION
**Pull request by derrikzzz**
Closes #120 

## Description

- Add unit tests for the `VitalsForm` covering: loading state, empty form field rendering, pre-filling with existing vitals data, success toast on save, and info toast when no fields are changed
- Fix the "no fields changed" notification from toast.error to a neutral toast (info style), rationale is that this should not be considered as an error condition and this is merely informing the user that no fields are changed

## Test plan
- [ ] Run `npx vitest run src/__tests__/pages/vitals` — all of the 5 tests should pass
- [ ] Manually open the vitals page, select a visit, and click "Save Records" without changing anything — confirm the toast appears as a neutral notification (not error toast)

### CODE REVIEW

File: src/__tests__/pages/vitals/[id]/index.test.tsx
Comment (naa-siuuuu-ff) [inline src/__tests__/pages/vitals/[id]/index.test.tsx:1]: LGTM! All tests run successfully

**Review by naa-siuuuu-ff — APPROVED** [review]

File: src/__tests__/pages/vitals/[id]/index.test.tsx [hunk 1]
```diff
@@ -1,38 +1,205 @@
...
+      <Toaster />
+      <PatientVitalsPage />
+    </>,
+  );
+}
+
+describe("VitalsForm", () => {
   beforeEach(() => {
     vi.clearAllMocks();
     toast.dismissAll();
```
Comment (keiran0) [inline src/__tests__/pages/vitals/[id]/index.test.tsx]: Should this be replaced with `toast.remove` instead? As per [documentation](https://react-hot-toast.com/docs/toast), `toast.dismiss` plays animations which may interfere with other tests. I've seen it happen in other tests that I've written where multiple toasts may show up leading to tests failing due to animations taking time to remove toasts from the screen (test leakage).

File: src/__tests__/pages/vitals/[id]/index.test.tsx [hunk 2]
```diff
@@ -1,38 +1,205 @@
...
+  test("shows info toast when submitting without changing any field", async () => {
+    const user = userEvent.setup();
+    mockTrpc.vitalsRouter.getByVisitId.useQuery.mockReturnValue({
+      data: null,
+      isLoading: false,
+    });
+
+    renderPage();
+
+    await waitFor(() => {
```
Comment (keiran0) [inline src/__tests__/pages/vitals/[id]/index.test.tsx:197]: While it seems unlikely for another instance of 'Save Records' text to be on the screen (other than the button), should this interactable part be more explicitly targeted? The toast can also be targeted using the `status` role.

```
const saveRecordsButton = screen.getByRole("button", { name: "Save Records"});
    expect(saveRecordsButton).toBeInTheDocument();
    await user.click(saveRecordsButton);

    const toast = screen.getByRole("status"); //
    expect(toast).toBeInTheDocument()
    expect(toast.textContent).toBe("No form field changed!");
```

File: src/__tests__/pages/vitals/[id]/index.test.tsx [hunk 3]
```diff
@@ -1,38 +1,205 @@
...
+    mockTrpc.vitalsRouter.create.useMutation.mockImplementation(
+      ({ onSuccess }: { onSuccess: () => void }) => ({
+        mutate: vi.fn(() => onSuccess()),
+        isPending: false,
+      }),
+    );
+
+    renderPage();
+
+    await waitFor(() => {
```
Comment (keiran0) [inline src/__tests__/pages/vitals/[id]/index.test.tsx:173]: Would it be better to target screen content by roles? (duplicate of my other review comment)

```   
 expect(await screen.findByLabelText("Height (cm)")).toBeInTheDocument();

    // Dirty the form so isDirty guard doesn't block submission, value here to be different from what is in MOCK_VITALS
    await user.type(screen.getByLabelText("Height (cm)"), "171");
    await user.click(screen.getByRole("button", { name: "Save Records" }))

    const toast = screen.getByRole("status");
    expect(toast).toBeInTheDocument();
    expect(toast.textContent).toBe("Vitals saved successfully!")
```

**Review by keiran0 — CHANGES_REQUESTED** [review]
Tests work as expected, but perhaps some improvement can be done by targeting roles instead of casting a wide net for text content. (Though note that the proposed improvements would require `toast.remove` instead of `toast.dismiss` because the animations interfere with each other causing failing tests due to multiple statuses appearing on the screen from other tests)

File: src/__tests__/pages/vitals/[id]/index.test.tsx (hunk 1 above)
Comment (derrikzzz) [reply src/__tests__/pages/vitals/[id]/index.test.tsx]: Good catch, have updated to use `toast.remove()` to remove animations immediately and prevent leakage in [039a3d5](https://github.com/NUS-Project-SaBai/project-sabai/pull/121/commits/039a3d523e63e20470bc3a999ba74f1f0ed5d55a)

File: src/__tests__/pages/vitals/[id]/index.test.tsx (hunk 2 above)
Comment (derrikzzz) [reply src/__tests__/pages/vitals/[id]/index.test.tsx:197]: Oh I guess that works to on using `getByRole`, have updated in [039a3d5](https://github.com/NUS-Project-SaBai/project-sabai/pull/121/commits/039a3d523e63e20470bc3a999ba74f1f0ed5d55a)

File: src/__tests__/pages/vitals/[id]/index.test.tsx (hunk 3 above)
Comment (derrikzzz) [reply src/__tests__/pages/vitals/[id]/index.test.tsx:173]: Have updated for more explicit targeting to target element in [039a3d5](https://github.com/NUS-Project-SaBai/project-sabai/pull/121/commits/039a3d523e63e20470bc3a999ba74f1f0ed5d55a)

**Review by keiran0 — APPROVED** [review]
LGTM!
